litani

```diff
@@ -416,6 +421,12 @@ def non_negative_int(arg):
     return ret
 
 
+def _non_directory_path(arg):
+    if pathlib.Path(arg).exists() and pathlib.Path(arg).resolve().is_dir():
+        raise ValueError("--out-file flag expects a file and not a directory")
+    return pathlib.Path(arg).resolve()
```

Comment:
I don't think we should resolve. If the user gave us a symlink, we should write to the symlink handle.